substrate_client/src/lib.rs

```diff
@@ -0,0 +1,70 @@
+use std::{str::FromStr, sync::Arc};
+
+use anyhow::{Ok, Result, Context};
+use codec::{Decode, Encode};
+use sp_core::{ed25519};
+use substrate_api_client::Api;
+
+
+#[derive(Clone)]
+pub struct SubstrateClient
+{
+    api: Arc<Api<ed25519::Pair>>,
+}
+
+impl SubstrateClient
+{
+    pub fn new(url: String) -> Result<Self> {
+        let api = Arc::new(Api::<ed25519::Pair>::new(url).context("failed to create substrate client")?);
+        Ok(Self { api })
+    }
+
+    pub fn get_twin<T: Decode>(&self, id: u32) -> Result<T> {
+        let twin: T = self
+            .api
+            .get_storage_map("TfgridModule", "Twins", id.encode(), None)?
+            .ok_or(anyhow::anyhow!("twin id is not found"))?
+            .decode()?;
+
+        Ok(twin)
+    }
+
+    pub fn get_twin_id_by_account_id(&self, account_id: String) -> Result<u32> {
+        let account_id = sp_core::ed25519::Public::from_str(&account_id)
+            .map_err(|err| anyhow::anyhow!(format!("{:?}", err)))?;
```

Comment:
Why do u do `map_err` and not just `?`
```suggestion
        let account_id = sp_core::ed25519::Public::from_str(&account_id).context("failed to get ed25519 key from account id")?;
```

Reply:
I remember we talked about that and tried to change it but we can't change it because the anyhow::Result has traits not implemented for std Result